pkg/util/cgroups/reader.go

```diff
@@ -188,6 +189,19 @@ func (r *Reader) GetCgroup(id string) Cgroup {
 	return r.cgroups[id]
 }
 
+// GetCgroupByInode returns the cgroup for the given inode.
+func (r *Reader) GetCgroupByInode(inode uint64) (Cgroup, error) {
+	r.cgroupsLock.RLock()
+	defer r.cgroupsLock.RUnlock()
+
+	cgroup, ok := r.cgroupByInode[inode]
+	if !ok {
+		return nil, fmt.Errorf("cgroup not found for inode %d", inode)
```

Comment:
You need to be aligned with `GetCgroup` behaviour. Not found is not an error, we just return `nil` Cgroup.

Reply:
Alright, I changed the function definition to respect that